packages/indexer/src/indexers/amm-market/amm-market-account-indexer.ts

```diff
@@ -0,0 +1,69 @@
+import { AccountInfoIndexer } from "../account-info-indexer";
+import { rpcReadClient } from "../../connection";
+import { AccountInfo, Context, PublicKey } from "@solana/web3.js";
+import { schema, usingDb } from "@metadaoproject/indexer-db";
+import { Err, Ok } from "../../match";
+import { BN } from "@coral-xyz/anchor";
+
+type TwapRecord = typeof schema.twaps._.inferInsert;
+
+export enum AmmAccountIndexerError {
+  GeneralError = "GeneralError",
+}
+
+// Doing this rather than class implements pattern due to
+// https://github.com/microsoft/TypeScript/issues/41399
+export const AmmMarketAccountUpdateIndexer: AccountInfoIndexer = {
+  index: async (
+    accountInfo: AccountInfo<Buffer>,
+    account: PublicKey,
+    context?: Context
+  ) => {
+    try {
+      const ammMarketAccount = await rpcReadClient.markets.amm.decodeMarket(
+        accountInfo
+      );
+      //index twap
+
+      // agg is 0 so skipping
+      if (ammMarketAccount.oracle.aggregator.toNumber() === 0)
+        return Ok({ acct: account.toString() });
+
+      const twapCalculation: BN = ammMarketAccount.oracle.aggregator.div(
+        ammMarketAccount.oracle.lastUpdatedSlot.sub(
+          ammMarketAccount.createdAtSlot
+        )
+      );
+      const twapNumber: number = twapCalculation.toNumber();
+      const newTwap: TwapRecord = {
+        curTwap: BigInt(twapNumber),
+        marketAcct: account.toString(),
+        observationAgg: ammMarketAccount.oracle.aggregator
+          .toNumber()
+          .toString(),
+        proposalAcct: ammMarketAccount.proposal.toString(),
+        // alternatively, we could pass in the context of the update here
+        updatedSlot: context
+          ? BigInt(context.slot)
+          : BigInt(ammMarketAccount.oracle.lastUpdatedSlot.toNumber()),
+        createdAt: new Date(),
```

Comment:
I'm a bigger proponent of using the DBs default vs JS stuff, NOW() is a perfectly great system.

Reply:
hmmm okay. I guess I assumed drizzle does the conversion. Let me see if drizzle specifies their own form of "now()".

Reply:
ohhh wait... so we just don't fill this in and it will work I guess then right? Because it has a default set?

Reply:
yep got to it after this comment...

Reply:
Made the change here.